TYPO3.Flow/Classes/TYPO3/Flow/Persistence/Doctrine/EntityManagerFactory.php

```diff
@@ -97,6 +97,17 @@ public function create()
         $config->setProxyNamespace('TYPO3\Flow\Persistence\Doctrine\Proxies');
         $config->setAutoGenerateProxyClasses(false);
 
+        // Set default host to 127.0.0.1 if there is no other host configured and at least one other necessary option
+        if ($this->settings['backendOptions']['host'] == '') {
+            if (
+                $this->settings['backendOptions']['dbname'] != '' ||
+                $this->settings['backendOptions']['user'] != '' ||
+                $this->settings['backendOptions']['password'] != ''
+            ) {
+                $this->settings['backendOptions']['host'] = '127.0.0.1';
+            }
+        }
+
```

Comment:
at least the code needs to be more robust, this could easily lead to issues with missing array keys and unsetting to null instead of an empty string..

Example:
```
if (empty($this->settings['backendOptions']['host']) &&
  (
    !empty($this->settings['backendOptions']['dbname']) ||
    !empty($this->settings['backendOptions']['user']) ||
    !empty($this->settings['backendOptions']['password'])
  )
) {
  $this->settings['backendOptions']['host'] = '127.0.0.1';
}